oshi-core/src/main/java/oshi/software/common/AbstractNetworkParams.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Oshi (https://github.com/dblock/oshi)
+ *
+ * Copyright (c) 2010 - 2017 The Oshi Project Team
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Maintainers:
+ * dblock[at]dblock[dot]org
+ * widdis[at]gmail[dot]com
+ * enrico.bianchi[at]gmail[dot]com
+ *
+ * Contributors:
+ * https://github.com/dblock/oshi/graphs/contributors
+ */
+package oshi.software.common;
+
+import com.sun.jna.ptr.PointerByReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.List;
+
+import oshi.jna.platform.unix.LibC;
+import oshi.software.os.NetworkParams;
+import oshi.util.FileUtil;
+
+/**
+ * Common NetworkParams implementation.
+ */
+public abstract class AbstractNetworkParams implements NetworkParams {
+
+    private static final long serialVersionUID = 1L;
+
+    private static final Logger LOG = LoggerFactory.getLogger(AbstractNetworkParams.class);
+    private static final String NAMESERVER = "nameserver";
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getDomainName() {
+        LibC.Addrinfo hint = new LibC.Addrinfo();
```

Comment:
These `LibC` calls here are specific to Unix, and this code will only work on FreeBSD and Solaris.  Either this Unix-specific should be implemented in a Unix class that accept the unix lass you imported, or you should do an OS check and import the appropriate LibC to use (the Linux LibC is a different class, and SystemB is the equivalent for the OS X case).